configuration/doc/README.md

```diff
@@ -27,11 +27,121 @@ plain backend, which only tracks calls to the compiler, but does not act upon
 them. An example invocation for the plain backend and a project whose build
 command is "make" would be:
 
- one-line-scan --plain --no-gotocc -- make
+  one-line-scan --plain --no-gotocc -- make
 
-More invocations of one-line-scan are presented in the file
-configuration/doc/EXAMPLES.md
+The available parameters of the tool can be seen when using the --help flag:
+
+  one-line-scan --help
+
+More invocations of one-line-scan are presented in the file doc/EXAMPLES.md
+
+By default, the wrapper for CBMC is enabled. Furthermore, analysis is enabled by
+default. To disable CBMC, use --no-gotocc, and to disable analysis, use the
+parameter --no-analysis.
+
+## How It Works
+
+The tool one-line-scan places wrappers for common compilers like gcc, clang, as
+well as g++ and clang++ in a new directory. If the parameter --suffix $SUFFIX
+or --prefix $PREFIX is used, the wrapper is also created for binaries of the
+name ${PREFIX}compiler and compiler${SUFFIX}. This allows to also track compiler
+like gcc-9, or x86_64-unknown-linux-gnu-gcc.
+
+After all wrappers have been placed in the directory, the environment variable
+PATH is prepended with this directory. Hence, any attempt to spot the compiler
+via the PATH variable will result in picking up the wrapper, instead of the
+actual compiler. This allow us the take action on compiler invocations.
+
+To check which compiler would be visible inside the build command, run the
+following command that uses the plain wrapper:
+
+  one-line-scan --plain --no-gotocc -o OLS --trunc-existing -- which gcc
+
+With this mechanism, compiler calls based on their absolute path cannot be
+intercepted. Consequently, analysis will not detect these calls. To still be
+able to get these calls, one-line-scan should be used during the configuration
+with "./configure" or "cmake", and next should re-use the created directory for
+the actual compilation. This way, the wrapper will be used during configuration.
+See call examples in configuration/doc/EXAMPLES.md or more details below.
+
+## How to Use For
+
+### Different Build Systems
+
+This section explains the usage of one-line scan for the build systems using
+plain compilation, make, configure and make, as well as cmake, and give a brief
+example on what should be done.
```

Comment:
Instead of "cmake, and give a brief example on what should be done." I'd say: "cmake. For each of them, a brief example is provided."